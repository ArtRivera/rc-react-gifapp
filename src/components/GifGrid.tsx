import React, { useState, useEffect } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
interface Props{
    category: any
}
const GifGrid = ({category}: Props) => {

    const {data: images ,loading} = useFetchGifs(category);

    return (
        <>
        <h1>{category}</h1>
        {loading && <p>Loading...</p>}
        <div className="card-grid">
               {images.map((i:any) => <GifGridItem key={i.id} img={{...i}}/>)}
        </div> 
        </>
    )
}

export default GifGrid
