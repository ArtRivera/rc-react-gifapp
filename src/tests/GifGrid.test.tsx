 import './setupTests';
 import { shallow } from 'enzyme'
 import GifGrid from '../components/GifGrid';
 import {useFetchGifs} from '../hooks/useFetchGifs';
 jest.mock('../hooks/useFetchGifs');
 
 describe('Pruebas de <GifGrid />', () => {

    const category = 'Naruto';

     test('should match to snapshot', () => {
         const wrapper = shallow(<GifGrid category={category}/>)
         expect(wrapper).toMatchSnapshot();
     })

     test('should mostrar items cuando se cargan de useFetchGifs', () => {
         const gifs =[{
            id: 'fvsvfv',
            title: 'sfdvvdvdf',
            url: 'dfvdfvsdvdsdf'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItems').length).toBe(gifs.length);
        
     })
     
     
 })
 