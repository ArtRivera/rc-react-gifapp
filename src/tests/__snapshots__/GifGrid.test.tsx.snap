// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid /> should match to snapshot 1`] = `
<Fragment>
  <h1>
    Naruto
  </h1>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItems
      img={Object {}}
    />
  </div>
</Fragment>
`;
